src/test/java/io/stargate/sgv2/jsonapi/api/v1/FindIntegrationTest.java

```diff
@@ -1727,6 +1727,220 @@ public void NeSubdoc() {
+          .contentType(ContentType.JSON)
+          .body(json)
+          .when()
+          .post(CollectionResource.BASE_PATH, namespaceName, collectionName)
+          .then()
+          .statusCode(200)
+          .body("status", is(nullValue()))
+          .body("errors", is(nullValue()))
+          .body("data.documents", hasSize(5))
+          .body(
+              "data.documents",
+              containsInAnyOrder(
+                  jsonEquals(expected2),
+                  jsonEquals(expected3),
+                  jsonEquals(expected4),
+                  jsonEquals(expected5),
+                  jsonEquals(expected6)));
+    }
+
+    @Test
+    public void nestedNotPushDown() {
+      String json =
+          """
+                {
+                    "find": {
+                        "filter": {
+                            "$not":
+                            {
+                              "$and": [
+                                {
+                                    "$or": [
+                                        {
+                                            "username": "user3"
+                                        },
+                                        {
+                                            "subdoc.id": {
+                                                "$eq": "abc"
+                                            }
+                                        }
+                                    ]
+                                },
+                                {
+                                    "$or": [
+                                        {
+                                            "username": "user2"
+                                        },
+                                        {
+                                            "subdoc.id": {
+                                                "$eq": "xyz"
+                                            }
+                                        }
+                                    ]
+                                }
+                              ]
+                            }
+                        }
+                    }
+                }
+                  """;
+
+      String expected =
+          """
+                      {"_id":"doc2", "username":"user2", "subdoc":{"id":"abc"},"array":["value1"]}
+                      """;
+      given()
+          .header(HttpConstants.AUTHENTICATION_TOKEN_HEADER_NAME, getAuthToken())
+          .contentType(ContentType.JSON)
+          .body(json)
+          .when()
+          .post(CollectionResource.BASE_PATH, namespaceName, collectionName)
+          .then()
+          .statusCode(200)
+          .body("status", is(nullValue()))
+          .body("errors", is(nullValue()))
+          .body("data.documents", not(containsInAnyOrder(expected)))
+          .body("data.documents", hasSize(5));
+    }
+
+    @Test
+    public void multipleNotCancelling() {
```

Comment:
Nice, I'm glad you tested this. I checked whether multiple `$not` works in MongoDB and Mongoose: MongoDB supports it, but Mongoose doesn't. Something for Mongoose to fix.